core/math/plane.cpp

```diff
@@ -135,7 +135,7 @@ bool Plane::intersects_segment(const Vector3 &p_begin, const Vector3 &p_end, Vec
 	real_t den = normal.dot(segment);
 
 	//printf("den is %i\n",den);
-	if (Math::abs(den) <= CMP_EPSILON) {
+	if (!Math::is_zero_approx(den)) {
```

Comment:
That changes the logic, it should not be negated.

Reply:
There's also the same original check in `intersects_ray` BTW, might be worth changing too (was probably not there when you did the initial implementation).

Reply:
Good catches! Fixed.